backend/ccm/canvas_api/instructor_sections_api_handler.py

```diff
@@ -0,0 +1,75 @@
+import logging
+from http import HTTPStatus
+from canvasapi import Canvas
+from rest_framework.views import APIView
+from rest_framework import authentication, permissions
+from rest_framework_tracking.mixins import LoggingMixin
+from rest_framework.response import Response
+from rest_framework.request import Request
+
+from canvasapi.exceptions import CanvasException
+from backend.ccm.canvas_api.canvas_credential_manager import CanvasCredentialManager
+from backend.ccm.canvas_api.canvasapi_serializer import CanvasObjectROSerializer
+from backend.ccm.canvas_api.course_section_api_handler import CanvasCourseSectionAPIHandler
+from backend.ccm.canvas_api.exceptions import CanvasErrorHandler, HTTPAPIError
+
+logger = logging.getLogger(__name__)
+
+class CanvasInstructorSectionsAPIHandler(LoggingMixin, APIView):
+    """
+    API handler for "merge-able" sections data for users with instructor-level access
+    """
+    logging_methods = ['GET']
+    authentication_classes = [authentication.SessionAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+    courses_allowed_fields = {"id", "name", "enrollment_term_id"}
+    sections_allowed_fields = {"id", "name", "course_id", "nonxlist_course_id", "total_students"}
+
+    def __init__(self, credential_manager=None):
+        self.credential_manager = credential_manager or CanvasCredentialManager()
+        self.canvas_error = CanvasErrorHandler()
+        super().__init__()
+
+    def get(self, request: Request) -> Response:
+        term_id = request.query_params.get('term_id')
+        if not term_id:
+            return Response({"error": "Term ID is required as a parameter"}, status=HTTPStatus.BAD_REQUEST)
+        try:
+            canvas_api = self.credential_manager.get_canvasapi_instance(request)
```

Comment:
this doesn't need to be part of try-catch, as this is redirect to drf exception handler when exception is raised. please look in other `*_handler.py` for similar patterns